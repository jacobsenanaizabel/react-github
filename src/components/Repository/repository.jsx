import React from 'react'
import axios from 'axios'

import './repository.scss'

export default class Repository extends React.Component{

    constructor(props) {
        super(props);
        this.state = {
            repositoryListReturn : [],
            repositoryList : this.props.location.state.repository,
            order:''
        }
    }

          
    handleChange(value){
        this.setState({
            order: value
        });

        console.log(this.state.repositoryListReturn);
    }

    sortDecrescent(){

    }

    sortCrescent(){

    }

    componentDidMount() {
        axios.get(this.state.repositoryList, { crossdomain: true } )
            .then(response => {
                this.setState({repositoryListReturn : response.data});
            }
        )
    }

    render() {
        return(     
            <div className="repository-container">
                <div className="repository-header">
                    <h3 className="title-repository"> Repositorios </h3>
                    <div className="container-order">
                        <label className="title-order">Ordenado de forma </label>
                        <select className="select-order" value={this.state.order} onChange={e => this.handleChange(e.target.value)}>
                            <option value="cresc">Crescente</option>
                            <option value="decresc">Decrescente</option>
                        </select>
                    </div>
                </div>
                <div className="repository-body">
                    <ul>
                    { this.state.repositoryListReturn.map(repository => 
                        <li className="nav-repository">
                            <a href={repository.html_url} className="text-name"> {repository.name} </a>
                            <h3 ></h3>
                            <label><p className="description">{repository.description}</p></label>
                            <div className="technique-description">
                                <label className="language">Linguagem : {repository.language} </label>
                                <label className="stars">Estrelas : {repository.stargazers_count} </label>
                            </div>
                        </li>
                    )}
                    </ul>
                </div>
            </div>
        )
    }
}