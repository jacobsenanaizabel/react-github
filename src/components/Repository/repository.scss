.repository-container {
    background-color: #fafbfc;
    font-family: 'Roboto Condensed';
    margin-left: 18%;
    width: 60%;

    .repository-body {
        .nav-repository {
            border-bottom: 1px solid #beb9b9;

            .description {
                font-size: 16px;
            }

            .text-name {
                font-size: 25px;
            }

            .technique-description {
                color: #bab3b3;
            }            
        }
    }

    .repository-header {
        width: 100%;
        height: 72px;

        .container-order {
            float: right;
            margin-right: 5%;
            margin-top: 2%;

            label.title-order {
                padding: 5px;
            }
        }
        
        .title-repository {
            float: left;
            padding-left: 5%;
            padding-top: 2%;
        }
    }
}