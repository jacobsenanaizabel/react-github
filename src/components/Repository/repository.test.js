import React from 'react'
import { shallow, mount } from 'enzyme'
import Repository from './repository'

describe('Repository of a user', () =>{
    it('should be return repositories', () =>{
        const wrapper = shallow(<Repository/>)

        const repository = wrapper.state().repositoryListReturn
        expect(repository).toBeDefined
    })

})