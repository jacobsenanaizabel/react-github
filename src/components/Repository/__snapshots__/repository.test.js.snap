// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repository /> renders repository snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "jwqpl7",
          "pathname": "/repository",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "jwqpl7",
        "pathname": "/repository",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Repository />
</Router>
`;
